jobspec/parse.go

```diff
@@ -619,6 +619,10 @@ func parseResources(result *structs.Resources, list *ast.ObjectList) error {
 		result.Networks = []*structs.NetworkResource{&r}
 	}
 
+	// Combine the parsed resources with a default resource block.
+	min := structs.DefaultResources()
```

Comment:
I think this logic should be in the jobs endpoint too? Users who are submitting the jobs via the http API can optionally get the default resources if they don't specify